code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -251,8 +251,8 @@
 ///For weird backwards situations where water manages to get added to trays nutrients, as opposed to being snowflaked away like usual.
 /datum/reagent/water/on_hydroponics_apply(obj/item/seeds/myseed, datum/reagents/chems, obj/machinery/hydroponics/mytray)
 	. = ..()
-	// Chek if this is the exact same type
-	if(chems.has_reagent(src, 1))
+	// Chek if this is the exact same reagent
+	if(chems.has_reagent(/datum/reagent/water))
```

Comment:
yeah first it checks if it atleast has 1 unit, then we have reagents with subtypes where we dont want the effects to be applied, like holywater or radium, so it does check if we have any amount of that exact same reagent since we already checked if it has atleast 1 unit. I didnt wanted to override it with return since then it would give some overhead, I guess I could comment it better out

Reply:
oh you meant chem.has_reagent(src, 1) or chem.has_reagent(chem, 1) in all cases those checks would end up becoming false, they were used ever since the botany rework and I dont know why.